3 reports.sql

```diff
@@ -0,0 +1,57 @@
+use HighSchoolReportsDB
+go
+
+--1) List all students that have any failing test scores. Sort by Grade(9-12), then Class, then Last Name
+select *
+from ReportCard r
+where MathScore < 65 or ScienceScore < 65  or HistoryScore < 65 or EnglishScore < 65
+order by r.Grade, r.ClassroomNumber, r.LastName
+
+--2) List average for each subject, per grade(9-12).
+select r.Grade, MatchAvg = AVG(r.MathScore), ScienceAvg = AVG(r.ScienceScore), HistoryAvg = AVG(r.HistoryScore), EnglishAvg = AVG(r.EnglishScore)
+from ReportCard r
+group by r.Grade
+
+--3) Our school gives an award for the 4 students with the highest averages that also receive an S+ in Conduct. 
+--	 Please list the 4 highest averages in our school. (Can be more than 1 per grade)
+--	 Please include Name, Grade, Class, and all test scores in this report. 
+select top 4 
+	r.FirstName, 
+	r.LastName, 
+	r.Grade, 
+	r.ClassroomNumber, 
+	GeneralAvg = ((MathScore + ScienceScore + HistoryScore + EnglishScore) / 4), 
```

Comment:
if you add GeneralAvg as column, then you don't have to do it manually here or by rest of reports....